File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> strDict;

        for(string str: strs) {
            string temp = str;
            sort(temp.begin(), temp.end());

            if (!strDict.contains(temp)) {
                strDict[temp] = {};
            }

            strDict[temp].push_back(str);
        }
        vector<vector<string>> res;
        for(const auto& [key, value]: strDict) {
            vector<string> temp;
            for (const auto& v: value) {
                temp.push_back(v);
            }
            res.push_back(temp);
        }

        return res;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        nums.sort()

        def dfs(temp, i, total):
            if total == target:
                res.append(temp.copy())
                return

            if total > target or i >= len(nums):
                return

            for j in range(i, len(nums)):
                temp.append(nums[j])
                dfs(temp, j, total + nums[j])
                temp.pop()

        dfs([], 0, 0)
        return res